src/app/components/tables/data-table/table-row/table-row.component.spec.ts

```diff
@@ -0,0 +1,139 @@
+
+    beforeEach(async(() =>
+    {
+        TestBed.configureTestingModule({
+            declarations: [
+                TableRowComponent,
+                TerraCheckboxComponent,
+            ],
+            imports:      [
+                FormsModule,
+            ],
+            providers:    [
+                {
+                    provide:  TerraDataTableComponent,
+                    useValue: dataTableStub
+                }
+            ]
+        }).compileComponents();
+    }));
+
+    beforeEach(() =>
+    {
+        fixture = TestBed.createComponent(TableRowComponent);
+        component = fixture.componentInstance;
+        dataTable = TestBed.get(TerraDataTableComponent);
+        fixture.detectChanges();
+    });
+
+    it('should create', () =>
+    {
+        expect(component).toBeTruthy();
+    });
+
+    it('should call data table #rowClicked() on click', () =>
+    {
+        const rowClicked:Spy = spyOn(dataTable, 'rowClicked');
+
+        fixture.debugElement.triggerEventHandler('click', {});
+
+        expect(rowClicked).toHaveBeenCalled();
+    });
+
+    it('should set classes according to rowData', () =>
+    {
+        let rowData:TerraDataTableRowInterface<any> = {
+            isActive: false,
+            selected: false,
+            disabled: false
+        };
+
+        component.row = rowData;
+
+        fixture.detectChanges();
+
+        expect(fixture.debugElement.classes['selected']).toBe(false);
+        expect(fixture.debugElement.classes['isActive']).toBe(false);
+        expect(fixture.debugElement.classes['disabled']).toBe(false);
+
+        rowData = {
+            isActive: true,
+            selected: true,
+            disabled: true
+        };
+
+        component.row = rowData;
+
+        fixture.detectChanges();
+
+        expect(fixture.debugElement.classes['selected']).toBe(true);
+        expect(fixture.debugElement.classes['isActive']).toBe(true);
+        expect(fixture.debugElement.classes['disabled']).toBe(true);
+    });
+
+    it('should toggle checkbox visibility', () =>
+    {
+        expect(fixture.debugElement.query(By.css('terra-checkbox'))).toBeTruthy();
+
+        dataTable.inputHasCheckboxes = false;
+
+        fixture.detectChanges();
+
+        expect(fixture.debugElement.query(By.css('terra-checkbox'))).toBeFalsy();
+    });
+
+    it('should call #onRowCheckboxChange() when checkbox changes', () =>
+    {
+        let rowData:TerraDataTableRowInterface<any> = {
+            isActive: false,
+            selected: false,
+            disabled: false
+        };
+
+        const onRowCheckboxChange:Spy = spyOn(dataTable, 'onRowCheckboxChange');
+
+        component.row = rowData;
+
+        fixture.detectChanges();
+
+        let checkbox:DebugElement = fixture.debugElement.query(By.css('terra-checkbox'));
+
+        checkbox.triggerEventHandler('change', {});
+
+        fixture.detectChanges();
+
+        expect(onRowCheckboxChange).toHaveBeenCalled();
```

Comment:
```suggestion
        expect(onRowCheckboxChange).toHaveBeenCalledWith(rowData);
```
Sonst würde die Zuweisung oben keinen Zweck haben ;-)